File: PracticeProblems/Leetcode/14.LongestCommonPrefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        min_len = float('inf')
        
        if len(strs) == 0: return ""
        if len(strs) == 1: return strs[0]
        
        for word in strs:
            if word == "": return ""
            if len(word) < min_len: min_len = len(word)

        for j in range(min_len):
            letter = [word[j] for word in strs]
            if len(set(letter)) != 1: return strs[0][0:j]

        return strs[0][0:min_len]


solution = Solution()
print(solution.longestCommonPrefix(["flower", "flow", "flight"]))
print(solution.longestCommonPrefix(["dog", "racecar", "car"]))
print(solution.longestCommonPrefix(["flower"]))
print(solution.longestCommonPrefix(["aa", "a"]))
print(solution.longestCommonPrefix([]))

File: practice/leetcode-js/14.LongestCommonPrefix.js
// Write a function to find the longest common prefix string amongst an array of strings.
// If there is no common prefix, return an empty string "".

const longestCommonPrefix = (strs) => {
  let prefix = "";
  let example = strs[0];
  for (let word of strs) {
    for (let i = 0; i < word.length; i++) {
      console.log(word[i]);
      if (example[i] !== word[i]) {
        prefix = example.slice(0, i);
        return prefix;
      }
    }
  }
  return prefix;
}

console.log(longestCommonPrefix(["flower", "flow", "flight"]));
console.log(longestCommonPrefix(["dog","racecar","car"]));
